sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -823,13 +823,18 @@ public Mono<MsalToken> authenticateWithBrowserInteraction(TokenRequestContext re
             return Mono.error(LOGGER.logExceptionAsError(new RuntimeException(e)));
         }
 
-        if (options.isBrokerEnabled() && options.useOperatingSystemAccount()) {
-            return getPublicClientInstance(request).getValue().flatMap(pc ->
+        // If the broker is enabled, try to get the token for the default account by passing
+        // a null account to MSAL. If that fails, show the dialog.
+        Mono<MsalToken> token = null;
+        if (options.isBrokerEnabled() && options.useDefaultBrokerAccount()) {
+            token = getPublicClientInstance(request).getValue().flatMap(pc ->
                 Mono.fromFuture(() ->
-                    acquireTokenFromPublicClientSilently(request, pc, null, false)).
+                        acquireTokenFromPublicClientSilently(request, pc, null, false)).
+                    // if something bad happened we fall back.
+                    onErrorResume(t -> Mono.empty()).
                     map(MsalToken::new));
-        } else {
-
+        }
+        if (token == null || token.hasElement().block() == Boolean.FALSE) {
```

Comment:
We shouldn't call .block() in an async API
Rather chain the entire flow reactively here

Reply:
Ah good catch I did this while I was testing and forgot to finish fixing it.